### PR TITLE
feat: Adding a connection test to postgres

### PR DESCRIPTION
**Pull request by mnorbury**
Implements the connection tester endpoint to report more detailed connectivity error messages back to the FE.

refs: https://github.com/cloudquery/cloudquery-issues/issues/1658

### CODE REVIEW

File: plugins/destination/postgresql/go.mod [hunk 1]
```diff
@@ -135,3 +135,7 @@ require (
 
 // github.com/cloudquery/jsonschema @ cqmain
 replace github.com/invopop/jsonschema => github.com/cloudquery/jsonschema v0.0.0-20240220124159-92878faa2a66
+
+replace github.com/cloudquery/plugin-sdk/v4 => github.com/disq/plugin-sdk/v4 v4.0.0-20240516105137-5410a74c47c5
+
+replace github.com/cloudquery/plugin-pb-go => github.com/cloudquery/plugin-pb-go v1.19.13-0.20240516105108-3fdf6ac15ea2
```
Comment (mnorbury) [inline plugins/destination/postgresql/go.mod:141]: To be removed before merge when `plugin-sdk` and `plugin-pb-go` are released.

File: plugins/destination/postgresql/client/connection_tester.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/rs/zerolog"
+)
+
+func ConnectionTester(ctx context.Context, _ zerolog.Logger, specBytes []byte) error {
+	var s spec.Spec
+	if err := json.Unmarshal(specBytes, &s); err != nil {
+		return &plugin.TestConnError{
+			Code:    plugin.TestConnFailureCodeUnknown,
```
Comment (disq) [inline plugins/destination/postgresql/client/connection_tester.go:21]:
```suggestion
			Code:    plugin.TestConnFailureCodeInvalidSpec,
```

File: plugins/destination/postgresql/client/connection_tester_test.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	tests := []struct {
+		name      string
+		specBytes []byte
+		wantErr   *wantErr
+	}{
+		{
+			name:      "should return an error for an invalid spec",
+			specBytes: []byte("invalid"),
+			wantErr: &wantErr{
+				Code:             plugin.TestConnFailureCodeUnknown,
```
Comment (disq) [inline plugins/destination/postgresql/client/connection_tester_test.go:40]:
```suggestion
				Code:             plugin.TestConnFailureCodeInvalidSpec,
```

**Review by savme — APPROVED** [review]

File: plugins/destination/postgresql/client/connection_tester.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	pgxConfig, err := pgxpool.ParseConfig(s.ConnectionString)
+	if err != nil {
+		return &plugin.TestConnError{
+			Message: processError(err),
+		}
+	}
+
+	c, err := pgxpool.NewWithConfig(context.Background(), pgxConfig)
```
Comment (disq) [inline plugins/destination/postgresql/client/connection_tester.go:38]: This can be `ctx`

**Review by disq — APPROVED** [review]

**Pull request merged**